### PR TITLE
electron-bin: rename from electron

### PR DESCRIPTION
**Pull request by prusnak**
###### Description of changes

Split out from https://github.com/NixOS/nixpkgs/pull/207766 to make that one easier to review

Supersedes https://github.com/NixOS/nixpkgs/pull/218406

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by prusnak** [comment]
Since this is based on your PR https://github.com/NixOS/nixpkgs/pull/218406 - can you please review @yu-re-ka?

**Review by AndersonTorres — APPROVED** [review]

**Review by yu-re-ka — APPROVED** [review]
Looks good. Thanks!

**Review by dotlambda — COMMENTED** [review]
Can we still move the expressions to `binary.nix` or a subfolder?

**Review by dotlambda — APPROVED** [review]

**Comment by prusnak** [comment]
> Can we still move the expressions to `binary.nix` or a subfolder?

Which expressions?

**Comment by dotlambda** [comment]
> > Can we still move the expressions to `binary.nix` or a subfolder?
> 
> Which expressions? Everything is in the `electron/binary` subfolder now.

I couldn't see that change on mobile. Once I realized I hid my comment.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -17170,23 +17170,40 @@ with pkgs;
...
+  electron_14 = electron_14-bin;
+  electron_15 = electron_15-bin;
+  electron_16 = electron_16-bin;
+  electron_17 = electron_17-bin;
+  electron_18 = electron_18-bin;
+  electron_19 = electron_19-bin;
+  electron_20 = electron_20-bin;
+  electron_21 = electron_21-bin;
+  electron_22 = electron_22-bin;
+  electron_23 = electron_23-bin;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:17206]: Can we move this into the electron file? I don't want to eventually have an alias jungle like java does.

**Pull request merged**